### PR TITLE
add extractionService for equations-to-model to use MIRA

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Option extractionService = mira | skema

see: https://github.com/DARPA-ASKEM/terarium/pull/5705

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java [hunk 1]
```diff
@@ -260,29 +261,54 @@ public ResponseEntity<UUID> equationsToModel(
...
-		// Get an AMR from Skema Unified Service
+		// Get an AMR from Skema Unified Service or MIRA
 		final Model responseAMR;
-		try {
-			responseAMR = skemaUnifiedProxy.consolidatedEquationsToAMR(skemaRequest).getBody();
-			if (responseAMR == null) {
-				log.warn("Skema Unified Service did not return a valid AMR based on the provided equations");
-				throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, messages.get("skema.bad-equations"));
+		String extractionService = "mira";
+		if (req.get("extractionService") != null) {
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java:267]: It feels really incorrect to have this be strongly typed on the front end in `knowledge.ts` and then have to deal with parsing json for values on the back end. This feels like something that should be generated from the back end and used from Types.ts

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java [hunk 2]
```diff
@@ -260,29 +261,54 @@ public ResponseEntity<UUID> equationsToModel(
...
+				taskReq.setUserId(currentUserService.get().getId());
+				final TaskResponse taskResp = taskService.runTaskSync(taskReq);
+				final JsonNode taskResponseJSON = mapper.readValue(taskResp.getOutput(), JsonNode.class);
+				final String amrString = taskResponseJSON.get("response").asText();
+				ObjectNode objNode = (ObjectNode) mapper.readTree(amrString);
+
+				final JsonNode testNode = mapper.readValue(amrString, JsonNode.class);
+				responseAMR = mapper.convertValue(testNode, Model.class);
+			} catch (Exception e) {
+				e.printStackTrace();
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java:287]: If we want a print out here it should use `log.error("some message", e);`

**Review by dvince2 — APPROVED** [review]
Consider turning the request into a typed object?

**Review by YohannParis — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java (hunk 1 above)
Comment (YohannParis) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java:267]: I agree with your comment, but this is a temporary solution for the hackathon, until we prove that MIRA is good enough and get rid of SKEMA.

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java (hunk 2 above)
Comment (blanchco) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java]: done!

**Pull request merged**